agent/xds/testdata/clusters/transparent-proxy-with-peered-upstreams.latest.golden

```diff
@@ -0,0 +1,97 @@
+{
```

Comment:
It looks like something may be missing here.

The listeners/transparent-proxy-with-peered-upstreams golden file reference a cluster with name `db.default.default.cloud.external.1c053652-8512-4373-90cf-5a7f6263a994.consul`. I would expect that same cluster to show up in these resources but I don't believe its in here.

Reply:
Any Listener or Route that points to a cluster must ultimately have that cluster emitted by CDS. Similarly if a cluster is configured and expects EDS, then the EDS endpoints should be emitted for that cluster else xDS effectively deadlocks waiting for a condition that will never resolve.

Reply:
Should we be emitting the no-endpoints listener, cluster and EDS endpoint info (with an empty set of endpoints). 

Per RBs comment above I _think_ that we should suppress emission of that somehow.

Reply:
If it works fine for tproxy already then I guess its fine. Maybe its only a deadlock if we never send EDS resources for the cluster which is subtly different from sending EDS resource with 0 endpoints defined.